packages/data-context/src/util/weightedChoice.ts

```diff
@@ -1,7 +1,7 @@
 import _ from 'lodash'
 
 export type WeightedAlgorithm = {
-  pick: Function
+  pick: (values: any[]) => string
```

Comment:
This is picking an element out of the `values` array and returning it, right? If so the types are incorrect here. Either should be `(values: string[]) => string` if it guarantees `string` or `<T>(values: T[]) => T` to allow returning whatever the type of the input array is. This would then flow down into your `weightedChoice` function below where there are a couple strict TS issues being disguised by the implicit `any` in play from this.

Reply:
Yes, values should be a `string[]`